cluster/connection_strategy_config.go

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cluster
+
+import (
+	"time"
+
+	"github.com/hazelcast/hazelcast-go-client/types"
+)
+
+type ConnectionStrategyConfig struct {
+	Retry            ConnectionRetryConfig
+	Timeout          types.Duration `json:",omitempty"`
+	DisableReconnect bool           `json:",omitempty"`
```

Comment:
Instead of a boolean, should we make this a separate type(ReconnectMode) with values 0 (OFF) and 1 (ON). This way, in the future, we could easily add the 2 (ASYNC) reconnect mode.

Reply:
APIs are sync-first in Go, so we probably won't have async mode for the Go client. The user can easily make async calls from sync ones.

Reply:
The async start might be an unfortunate name but basically, it means the client won't wait to connect to cluster in its start method. So, the method might be sync, but the behavior can be implemented any way

Reply:
Sorry, I mixed this up with the asyncStart boolean in the Java client in my previous comment. ASYNC reconnect mode is still a thing, so it might be good to do it anyways for the future

Reply:
Changed it at: https://github.com/hazelcast/hazelcast-go-client/pull/590/commits/1285e0da5b15d82430f385cf509e263f59c25ac5